File: back-end/src/main/java/es/ull/project/application/service/operation/OperationGetService.java
package es.ull.project.application.service.operation;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;

import es.ull.project.application.repository.OperationRepository;
import es.ull.project.application.usecase.operation.OperationGetUseCase;
import es.ull.project.domain.valueobject.Operation;
import es.ull.project.domain.valueobject.OperationName;

public class OperationGetService implements OperationGetUseCase {

    @Autowired
    private OperationRepository repository;

    @Override
    public Optional<Operation> getOperation(OperationName operationName) {
        return Optional.ofNullable(repository.fetchByName(operationName));
    }

    @Override
    public List<Operation> getAllOperations() {
        return repository.fetchAll();
    }
}

File: front-end/src/application/service/operation/operation-get-service.ts
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError, UllUUID } from '@ull-tfg/ull-tfg-typescript';
import type { OperationGetUseCase } from '../../usecase/operation/operation-get-use-case';
import { Operation } from '../../../domain/value-object/operation';
import type { OperationRepository } from '../../repository/operation-repository';
import { OperationName } from '@/domain/value-object/operation-name';

export class OperationGetService implements OperationGetUseCase {
    constructor(private readonly operationRepository: OperationRepository) {}
  
    async getOperation(operationName: OperationName): Promise<Either<DataError, Operation>> {
      return this.operationRepository.getOperation(operationName);
    }
  
    async getAllOperations(): Promise<Either<DataError, Operation[]>> {
      return this.operationRepository.getOperations();
    }
  }